crates/hyperdrive-math/src/lp.rs

```diff
@@ -353,6 +412,122 @@ impl State {
             ))
             .unwrap()
     }
+
+    /// Calculates the number of share reserves that are not reserved by open
+    /// positions.
+    fn calculate_idle_share_reserves(&self) -> FixedPoint {
+        let long_exposure = self.long_exposure().div_up(self.vault_share_price());
+        let idle_shares = {
+            if self.share_reserves() > long_exposure + self.minimum_share_reserves() {
+                return self.share_reserves() - long_exposure - self.minimum_share_reserves();
+            }
+            fixed!(0)
+        };
+
+        idle_shares
+    }
+
+    /// Given a signed bond amount, this function calculates the negation of the
+    /// derivative of `calculateSharesOutGivenBondsIn` when the bond amount is
+    /// positive or the derivative of `calculateSharesInGivenBondsOut` when the
+    /// bond amount is negative.
+    /// Note that the state is the present state of the pool and original values
+    /// passed in as parameters.  Present sate variables are not expressly
+    /// paased in because so that downstream function like kUp() can still be
+    /// used.
```

Comment:
This is missing the Ascii math from the contract code. There is other Ascii math in the function body that is missing. I think it would be best to just duplicate the comments since the documentation is good.